### PR TITLE
[Feature] Support Create Table Like DDL for Hive

### PR DESCRIPTION
**Pull request by 839224346**
What I'm doing:
Support Create Table Like DDL for Hive

## What type of PR is this:

- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [x] This pr needs user documentation (for new or modified features or behaviors)
- [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/server/MetadataMgr.java [hunk 1]
```diff
@@ -259,6 +260,47 @@ public boolean createTable(CreateTableStmt stmt) throws DdlException {
         }
     }
 
+    public void createTableLike(CreateTableLikeStmt stmt) throws DdlException {
+        String catalogName = stmt.getCatalogName();
+        Optional<ConnectorMetadata> connectorMetadata = getOptionalMetadata(catalogName);
+
+        if (connectorMetadata.isPresent()) {
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/server/MetadataMgr.java:267]: Analyze job has been done in the CreateTableLikeAnalyzer.

File: fe/fe-core/src/test/java/com/starrocks/server/MetadataMgrTest.java [hunk 2]
```diff
@@ -256,6 +266,112 @@ public void testCreateIcebergTable() throws Exception {
         Assert.assertFalse(metadataMgr.createTable(createTableStmt));
     }
 
+    @Test
```
Comment (stephen-shelby) [inline fe/fe-core/src/test/java/com/starrocks/server/MetadataMgrTest.java:269]: it's best to add sql test.

File: fe/fe-core/src/main/java/com/starrocks/server/MetadataMgr.java [hunk 3]
```diff
@@ -259,6 +260,47 @@ public boolean createTable(CreateTableStmt stmt) throws DdlException {
...
+            if (!CatalogMgr.isInternalCatalog(catalogName)) {
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/server/MetadataMgr.java:268]: I think internal catalog can also use this interface.

**Review by stephen-shelby — DISMISSED** [review]

**Review by packy92 — DISMISSED** [review]

**Review by stephen-shelby — DISMISSED** [review]

File: test/sql/test_ddl/T/create_table_like [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+`k1` DATE,
+`k2` DATETIME,
+`k3` VARCHAR(20),
+`k4` VARCHAR(20),
+`k5` BOOLEAN,
+`v1` TINYINT NULL,
+`v2` SMALLINT NULL,
+`v3` INT NULL,
+`v4` BIGINT NOT NULL,
+INDEX init_bitmap_index (k2) USING BITMAP
```
Comment (stephen-shelby) [inline test/sql/test_ddl/T/create_table_like:18]: why test the olap table?

File: test/sql/test_ddl/T/create_table_like [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+)
+DISTRIBUTED BY HASH(`k1`, `k2`, `k3`, `k4`, `k5`) BUCKETS 50
+ORDER BY(`k2`, `v4`)
+PROPERTIES (
+    "replication_num" = "1",
+    "storage_format" = "v2",
+    "enable_persistent_index" = "true",
+    "bloom_filter_columns" = "k1,k2"
+);
+CREATE TABLE test_pk_tbl2 LIKE test_pk_tbl1;
```
Comment (stephen-shelby) [inline test/sql/test_ddl/T/create_table_like:33]: you need to drop catalog/database/tables

Comment (839224346) [reply test/sql/test_ddl/T/create_table_like:33]: done

File: test/sql/test_ddl/T/create_table_like (hunk 4 above)
Comment (839224346) [reply test/sql/test_ddl/T/create_table_like:18]: UT covered Hive table and thought it would be nice to have the olap table covered as well

**Review by stephen-shelby — APPROVED** [review]

**Review by packy92 — APPROVED** [review]

**Comment by stephen-shelby** [comment]
@Mergifyio backport branch-3.2

**Pull request merged**